aten/src/ATen/native/MaxPooling.cpp

```diff
@@ -0,0 +1,126 @@
+#include <ATen/ATen.h>
+#include <ATen/NamedTensorUtils.h>
+#include <ATen/Parallel.h>
+#include <ATen/native/DispatchStub.h>
+#include <ATen/native/MaxPooling.h>
+
+namespace at {
+namespace native {
+
+DEFINE_DISPATCH(max_pool1d_stub);
+
+namespace {
+
+// Compute the output size for the given pooling parameters
+inline int64_t output_size(
+    int64_t input_size,
+    int64_t kernel_size,
+    int64_t stride,
+    int64_t padding,
+    int64_t dilation,
+    bool ceil_mode) {
+  int64_t num = input_size + 2 * padding - dilation * (kernel_size - 1) - 1;
+  // Ensure last kernel window starts within bounds in ceil mode
+  if (ceil_mode && stride - dilation * (kernel_size - 1) <= num % stride) {
```

Comment:
nit: I spend sometime to understand this. `stride - dilation * (kernel_size - 1) <= num % stride`, it would be better to add a little bit detailed comment here, something like:
// The method here is to calculate whether the last kernel move (by stride) will jump entirely out of bound